### PR TITLE
Add a service_catalog attribute to identity_auth_scope_v3 data

### PR DESCRIPTION
**Pull request by jeblair**
Keystone returns a service catalog along with the auth token.  That provides
useful information such as service endpoint URLs which may not otherwise be
available.  This change adds the service catalog to the
identity_auth_scope_v3 data source.

### CODE REVIEW

**Review by emonty — APPROVED** [review]
Exposing the service catalog is a great idea.

**Comment by jeblair** [comment]
Test failure is unrelated to this patch; tracking in https://github.com/theopenlab/openlab/issues/681

**Comment by jeblair** [comment]
recheck

File: openstack/identity_auth_scope_v3.go [hunk 1]
```diff
@@ -22,11 +22,37 @@ func flattenIdentityAuthScopeV3Roles(roles []tokens3.Role) []map[string]string {
 	return allRoles
 }
 
-func GetTokenDetails(sc *gophercloud.ServiceClient) (*tokens3.User, *tokens3.Domain, *tokens3.Project, []tokens3.Role, error) {
+func flattenIdentityAuthScopeV3ServiceCatalog(catalog tokens3.ServiceCatalog) []map[string]interface{} {
```
Comment (kayrus) [inline openstack/identity_auth_scope_v3.go:25]: can you pass the pointer here?
```suggestion
func flattenIdentityAuthScopeV3ServiceCatalog(catalog *tokens3.ServiceCatalog) []map[string]interface{} {
```

**Review by kayrus — CHANGES_REQUESTED** [review]
Thank you. This is a really nice feature. I'd appreciate if you add a couple of examples on how to use it in documentation examples.

File: openstack/data_source_openstack_identity_auth_scope_v3.go [hunk 2]
```diff
@@ -140,6 +189,13 @@ func dataSourceIdentityAuthScopeV3Read(d *schema.ResourceData, meta interface{})
 		log.Printf("[DEBUG] Unable to set openstack_identity_auth_scope_v3 roles: %s", err)
 	}
 
+	if catalog != nil {
+		flatCatalog := flattenIdentityAuthScopeV3ServiceCatalog(*catalog)
```
Comment (kayrus) [inline openstack/data_source_openstack_identity_auth_scope_v3.go:193]:
```suggestion
		flatCatalog := flattenIdentityAuthScopeV3ServiceCatalog(catalog)
```

File: openstack/identity_auth_scope_v3.go (hunk 1 above)
Comment (jeblair) [reply openstack/identity_auth_scope_v3.go]: Done.

File: openstack/data_source_openstack_identity_auth_scope_v3.go (hunk 2 above)
Comment (jeblair) [reply openstack/data_source_openstack_identity_auth_scope_v3.go]: Done.

**Comment by jeblair** [comment]
> Thank you. This is a really nice feature. I'd appreciate if you add a couple of examples on how to use it in documentation examples.

Of course.  The data structure isn't easy to navigate, and I'm new to HCL, so I don't know if there's a more efficient way to do it, but I added the code I'll be using for my use case, which is getting the public swift url.  Please let me know if there's a nicer way to do that; I won't be offended at all!  :)

**Comment by jeblair** [comment]
recheck

**Comment by jeblair** [comment]
recheck

**Comment by ozerovandrei** [comment]
Could you please rebase on main since there are fixes for acceptance tests.

**Comment by jeblair** [comment]
> Could you please rebase on main? There are fixes for acceptance tests.

Done!

**Review by ozerovandrei — APPROVED** [review]
LGTM

**Pull request merged**